backend/operations_scanner.go

```diff
@@ -481,6 +484,17 @@ func (s *OperationsScanner) pollClusterOperation(ctx context.Context, op operati
 	if err != nil {
 		var ocmError *ocmerrors.Error
 		if errors.As(err, &ocmError) && ocmError.Status() == http.StatusNotFound && op.doc.Request == database.OperationRequestDelete {
+			// Update the Cosmos DB billing document with a deletion timestamp.
+			// Do this before calling setDeleteOperationAsCompleted so that in
+			// case of error the backend will retry by virtue of the operation
+			// document still having a non-terminal status.
+			err = s.patchBillingDocument(ctx, op)
+			if err != nil {
+				s.recordOperationError(ctx, pollClusterOperationLabel, err)
+				op.logger.Error(fmt.Sprintf("Failed to handle a completed deletion: %v", err))
```

Comment:
Let's distinguish these so the poor sod trying to make sense of what failed, knows.

```suggestion
				op.logger.Error(fmt.Sprintf("Failed to update billing document after a completed deletion: %v", err))
```